resources/assets/js/photoAlbum.js

```diff
@@ -0,0 +1,315 @@
+            }
+        },
+        error: function (request, error) {
+            console.log(arguments);
+            alert(" Can't do because: " + error);
+        }
+    });
+}
+
+function addPhotoToAlbum(thumbnailResults, photoResults, index){
+    var i = index;
+    var j = 0;
+    if((thumbnailResults.length - index) <= 5){
+        j = (thumbnailResults.length - index);
+    } else{
+        j = 5;
+    }
+    var formData = new FormData();    
+    for(i; i < index+j; i++){
+        formData.append('thumbnails[]',thumbnailResults[i]);
+        formData.append('photos[]', photoResults[i]);
+    }
+    var url = currentPhotoAlbum;
+    var type = "POST";
+    formData.append("_token", window.Laravel.csrfToken);
+
+        $.ajax({
+            url: url,
+            data: formData,
+            type: type,
+            contentType: false,
+            processData: false,
+            success: function (result) {
+                count+=5;
+                if(count >= thumbnailResults.length){
+                    location.reload();
+                } else{
+                    addPhotoToAlbum(thumbnailResults, photoResults, i);
+                }
+            },
+            error: function (request, error) {
+                console.log(arguments);
+                alert(" Can't do because: " + error);
+            }
+    });
+    
+}
+
+function uploadPhoto() {
+    ChangebuttonState("Processing photos...")
+    var files = document.getElementById('file-select').files;
+    if(files.length != 0){
+        var albumName = $("input#inputTitle").val();
+        var albumDescription = $("textarea#textareaDescription").val();
+        var captureDate = $("input#CaptureDate").val();
+        var thumbnails = processThumbnails(files);
+        var photos = processPhotos(files);
+        
+        Promise.all(thumbnails).then(function(thumbnailResults){
+            Promise.all(photos).then(function(photosResults){
+                ChangebuttonState('Sending photos...')
+                if (albumName == undefined && albumDescription  == undefined) { // upload photo without album
+                    addPhotoToAlbum(thumbnailResults, photosResults, 0);
+                } else{
+                    if(albumName != '' && albumDescription  != ''){
+                        addAlbum(thumbnailResults, photosResults, albumName, albumDescription, captureDate);
+                    } else{
+                        var button = $('button#submit');
+                        button.html("Toevoegen");
+                        button.attr("disabled", false);
+                        alert("Voeg een album naam en beschrijving toe");
+                    } 
+                }
+            });
+
+        });
+    } else{
+        var button = $('button#submit');
+        button.html("Toevoegen");
+        button.attr("disabled", false);
+        alert("Selecteer een aantal foto's om toe te voegen."); 
+    }
+}
+
+function processPhotos(photos) {
```

Comment:
Deze functie geeft uiteindelijk een `Out of memory at ImageData creation` exception als je vrij veel foto's teglijk upload. (getest met ong 190) Ik stel voor om ofwel een limiet te stellen aan het aantal, of om het in chunks op te delen zodat niet alles in memory hoeft te blijven. Ook krijgt de user nu geen feedback dat er iets fout is gegaan, alleen een exception in de console.

Reply:
Ik heb hier even naar gekeken, en ik vindt een limiet erop zetten te makkelijk. 
Daarnaast dus even een performace record gedaan en het lijkt erop dat de downscaleCanvas functie de memoryleak is.(+/- 16 seconden per foto van 8 mb). 
![capture](https://user-images.githubusercontent.com/16254233/50670888-9e98bb80-0fce-11e9-83c7-9e838a9cf4ad.PNG)

Ben momenteel aan het kijken of dit op een betere manier kan om toch meerdere fotos te kunnen uploaden en de filesize wel gewenst te krijgen.